pkg/simple/client/monitoring/prometheus/prometheus.go

```diff
@@ -24,14 +25,35 @@ func NewPrometheus(options *Options) (monitoring.Interface, error) {
 	return prometheus{client: apiv1.NewAPI(client)}, err
 }
 
-// TODO(huanggze): reserve for custom monitoring
-func (p prometheus) GetMetrics(stmts []string, time time.Time) []monitoring.Metric {
-	panic("implement me")
+func (p prometheus) GetMetric(expr string, ts time.Time) monitoring.Metric {
+	var parsedResp monitoring.Metric
+
+	value, err := p.client.Query(context.Background(), expr, ts)
```

Comment:
context.Background or context.WithTimeout?
Is it reasonable to enforce the query timeout here?

Reply:
I think the context puts an extra requirement on the underlying HTTP connection.  But maybe we don't need to do that.

To enable a client-side timeout, we could adjust the http client on initialization. By now, I think the default setting is good.
https://github.com/kubesphere/kubesphere/blob/31135c17aea1cd76df037f4acc402b37bd9542be/pkg/simple/client/monitoring/prometheus/prometheus.go#L19-L26

Reply:
ok